extensions/replication/queueProcessor/MultipleBackendTask.js

```diff
@@ -362,21 +364,23 @@ class MultipleBackendTask extends QueueProcessorTask {
             Body: incomingMsg,
         });
         attachReqUids(destReq, log);
-        return destReq.send((err, data) => {
-            if (err) {
-                // eslint-disable-next-line no-param-reassign
-                err.origin = 'source';
-                log.error('an error occurred on putData to S3', {
-                    method: 'MultipleBackendTask._getAndPutPartOnce',
-                    entry: destEntry.getLogInfo(),
-                    origin: 'target',
-                    peer: this.destBackbeatHost,
-                    error: err.message,
-                });
-                return doneOnce(err);
-            }
-            sourceEntry.setReplicationDataStoreVersionId(data.versionId);
-            return doneOnce(null, data);
+        incomingMsg.once('readable', () => {
```

Comment:
`readable` event could be emitted multiple times. Were you not able to hook onto the `success` event?

Reply:
ya seemed like the `success` event was not being emitted. It looks like `incomingMsg` is an instance of `stream.PassThrough`.

I was thinking the `readable` event would be emitted because there was no error and instead data is available. Since `destReq.Body` accepts a stream, it was ok to `send` by that point? Purely assumptions though :) What do you think?

Reply:
Could you use the 'data' event to ensure that we can read from the stream?
http://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Request.html#createReadStream-property

Reply:
@bennettbuchanan I can try it out 👍

Reply:
`readable` with `once` handler should be sufficient. If you add a listener for `data` and ignore the chunk, the chunk is lost.